From 6694266a3ddc784c21c6fa7ee0ab57e8bc201e5a Mon Sep 17 00:00:00 2001
From: Boris Faure <billiob@gmail.com>
Date: Thu, 3 May 2018 21:41:28 +0200
Subject: fix focus issue when mouse moving over window without focusing

---
 src/bin/win.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bin/win.c b/src/bin/win.c
index 912981a..7a93e0a 100644
--- a/src/bin/win.c
+++ b/src/bin/win.c
@@ -1569,7 +1569,7 @@ _cb_win_mouse_move(void *data,
    Term_Container *tc = (Term_Container*) wn;
    Term_Container *tc_child = NULL;
 
-   if (wn->on_popover || wn->group_input)
+   if (wn->on_popover || wn->group_input || !tc->is_focused)
      return;
 
    if (!wn->config->mouse_over_focus)
-- 
cgit v0.12

