diff --git a/src/modules/lokker/lokker.c b/src/modules/lokker/lokker.c
index 82ea45d..26f940a 100644
--- a/src/modules/lokker/lokker.c
+++ b/src/modules/lokker/lokker.c
@@ -577,17 +577,6 @@ _lokker_cb_exit(void *data EINA_UNUSED, int type EINA_UNUSED, void *event)
         e_desklock_hide();
      }
    /* error */
-   else if ((!ev->exited) || (ev->exit_code < 128))
-     {
-        /* security - null out passwd string once we are done with it */
-        _lokker_null();
-        e_desklock_hide();
-        e_util_dialog_show(_("Authentication System Error"),
-                           _("Authentication via PAM had errors setting up the<ps/>"
-                             "authentication session. The error code was <hilight>%i</hilight>.<ps/>"
-                             "This is bad and should not be happening. Please report this bug.")
-                           , ev->exited ? ev->exit_code : ev->exit_signal);
-     }
    /* failed auth */
    else
      {
