--- a/media/audioipc/audioipc/src/cmsg.rs.orig	2018-05-10 13:40:31.504571013 +0200
+++ b/media/audioipc/audioipc/src/cmsg.rs	2018-05-10 13:41:04.818059434 +0200
@@ -108,6 +108,7 @@
 
             let cmsghdr = cmsghdr {
                 cmsg_len: cmsg_len as _,
+                __pad1: mem::size_of::<u64>() as i32 - mem::size_of::<i16>() as i32,
                 cmsg_level: level,
                 cmsg_type: kind,
             };
